crates/libs/bindgen/src/implements.rs

```diff
@@ -77,12 +77,13 @@ pub fn gen(def: &TypeDef, gen: &Gen) -> TokenStream {
         let signature = method.signature(&def.generics);
         let vtbl_signature = gen_vtbl_signature(def, &method, gen);
 
-        let invoke_upcall = if def.is_winrt() { gen_winrt_upcall(&signature, quote! { (*this).#name }) } else { gen_win32_upcall(&signature, quote! { (*this).#name }) };
+        let invoke_upcall = if def.is_winrt() { gen_winrt_upcall(&signature, quote! { this.#name }) } else { gen_win32_upcall(&signature, quote! { this.#name }) };
 
         quote! {
-            unsafe extern "system" fn #name<#(#constraints)* Identity: ::windows::core::IUnknownImpl, Impl: #impl_ident<#(#generics)*>, const OFFSET: isize> #vtbl_signature {
-                let this = (this as *mut ::windows::core::RawPtr).offset(OFFSET) as *mut Identity;
-                let this = (*this).get_impl() as *mut Impl;
+            unsafe extern "system" fn #name<#(#constraints)* Identity: ::windows::core::IUnknownImpl<Impl = Impl>, Impl: #impl_ident<#(#generics)*>, const OFFSET: isize> #vtbl_signature {
```

Comment:
Sure! So the `IUnknownImpl` trait now has an associated type with called `Impl` which is meant to be the implementation backing the COM object, and is what is returned from`IUnknownImpl::get_impl`. You can see the definition of that [here](https://github.com/microsoft/windows-rs/blob/b870059d2ae5e66df5b3088fd484c530670cd11f/crates/libs/windows/src/core/unknown.rs#L62-L64). 

The `Identity: IUnknownImpl<Impl = Impl>` (yes, that's a lot of impl) syntax means the generic parameter `Identity` must implement the trait `IUknownImpl` with an associated type that is itself the generic parameter `Impl`. So the `Impl = Impl` says "the associated type of the `IUnknownImpl` trait is called `Impl` and we're saying it must be whatever the generic parameter `Impl` is. 

This would probably be clearer if we used different names for the associated type and the generic parameter. Normally generic parameters are single letters like `T`, `U`, etc. so the syntax would end up looking like `Identity: IUnknownImpl<Impl = I>`.

Does that make sense? Any thoughts on how we can tweak names to make this clearer?

Reply:
[Here's a simplified version](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=ce11e7cd8ee3cb3725910a1d4a82798e) of what we're doing you can use to play around with how the syntax works.

Reply:
Got it - thanks! Yes, that's a lot of impl, but then this isn't something that devs need to interact with so perhaps not that important. I'm far more interested in find a way to drop the `Impl` from names like `IStringable_Impl` that devs need to utter...